packages/gitgraph-react/src/Commit.tsx

```diff
@@ -1,135 +1,180 @@
 import { Arrow } from "./Arrow";
 import { Message } from "./Message";
-import { Tag } from "./Tag";
+import { Tag, TAG_PADDING_X } from "./Tag";
 import { BranchLabel } from "./BranchLabel";
+import { MutableRefObject } from "react";
 
 interface CommitsProps {
-  commits: Array<Commit<ReactSvgElement>>;
-  commit: Commit<ReactSvgElement>;
-  currentCommitOver: Commit<ReactSvgElement> | null;
+  commits: Array<CommitCore<ReactSvgElement>>;
+  commit: CommitCore<ReactSvgElement>;
+  currentCommitOver: CommitCore<ReactSvgElement> | null;
   gitgraph: GitgraphCore<ReactSvgElement>;
-  initCommitElements: (commit: Commit<ReactSvgElement>) => void;
-  commitsElements: {
-    [commitHash: string]: CommitElement;
-  };
   getWithCommitOffset: (props: any) => Coordinate;
   setTooltip: (val: React.ReactElement<SVGGElement> | null) => void;
-  setCurrentCommitOver: (val: Commit<ReactSvgElement> | null) => void;
+  setCurrentCommitOver: (val: CommitCore<ReactSvgElement> | null) => void;
+  commitMessagesX: number;
 }
 
-export class CommitComp extends React.Component<CommitsProps, {}> {
-  public render() {
-    const commit = this.props.commit;
-    const { x, y } = this.props.getWithCommitOffset(commit);
-
-    const shouldRenderTooltip =
-      this.props.currentCommitOver === commit &&
-      (this.props.gitgraph.isHorizontal ||
-        (this.props.gitgraph.mode === Mode.Compact &&
-          commit.style.hasTooltipInCompactMode));
-
-    if (shouldRenderTooltip) {
-      this.props.setTooltip(
-        <g transform={`translate(${x}, ${y})`}>
-          <Tooltip commit={commit}>
-            {commit.hashAbbrev} - {commit.subject}
-          </Tooltip>
-        </g>,
-      );
-    }
+export const Commit = (props: CommitsProps) => {
+  const {commit, commits, gitgraph, commitMessagesX} = props;
 
-    return (
-      <g transform={`translate(${x}, ${y})`}>
-        <Dot
-          commit={commit}
-          onMouseOver={() => {
-            this.props.setCurrentCommitOver(commit);
-            commit.onMouseOver();
-          }}
-          onMouseOut={() => {
-            this.props.setCurrentCommitOver(null);
-            this.props.setTooltip(null);
-            commit.onMouseOut();
-          }}
-        />
-        {this.props.gitgraph.template.arrow.size && this.renderArrows(commit)}
-
-        {/* These elements are positionned after component update. */}
-        <g transform={`translate(${-x}, 0)`}>
-          {
-            commit.style.message.display &&
-            <Message
-              commit={commit}
-              commitsElements={this.props.commitsElements}
-              initCommitElements={this.props.initCommitElements}
-            />
-          }
-          {this.renderBranchLabels(commit)}
-          {this.renderTags(commit)}
-        </g>
-      </g>
-    );
-  }
+  // This _should_ likely be an array, but is not in order to intentionally keep
+  // a bug in the codebase that existed prior to Hook-ifying this component
```

Comment:
👍 For the comment, it will be helpful for maintenance.

Could you add a word about the actual bug? Is there an open issue we could link here if people are looking for context?